Microsoft.Toolkit.Win32/Microsoft.Toolkit.Forms.UI.Controls.WebView/WebViewCompatibilityAdapter.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Windows.Forms;
+using Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT;
+
+namespace Microsoft.Toolkit.Forms.UI.Controls
+{
+    internal sealed class WebViewCompatibilityAdapter : WebBaseCompatibilityAdapter, IWebViewCompatibleAdapter, IDisposable
+    {
+        private WebView _webView = new WebView();
+
+        ~WebViewCompatibilityAdapter()
+        {
+            Dispose(false);
+        }
+
+        public override event EventHandler<WebViewControlContentLoadingEventArgs> ContentLoading
+        {
+            add
+            {
+                _webView.ContentLoading += value;
+                _webView.FrameContentLoading += value;
+            }
+
+            remove
+            {
+                _webView.ContentLoading -= value;
+                _webView.FrameContentLoading -= value;
+            }
+        }
+
+        public override event EventHandler<WebViewControlNavigationCompletedEventArgs> NavigationCompleted
+        {
+            add
+            {
+                _webView.NavigationCompleted += value;
+                _webView.FrameNavigationCompleted += value;
+            }
+
+            remove
+            {
+                _webView.NavigationCompleted -= value;
+                _webView.FrameNavigationCompleted -= value;
+            }
+        }
+
+        public override event EventHandler<WebViewControlNavigationStartingEventArgs> NavigationStarting
+        {
+            add
+            {
+                _webView.NavigationStarting += value;
```

Comment:
How does a subscriber differentiate between navigation of the top level window and that of a frame?